lib/wibox/widget/imagebox.lua

```diff
@@ -253,10 +260,16 @@ function imagebox:draw(ctx, cr, width, height)
         if self._private.clip_shape then
             cr:clip(self._private.clip_shape(cr, w, h, unpack(self._private.clip_args)))
         end
+
+        if self._private.handle then
+            -- Translation is not needed in the new surface
+            svg_surf = cairo.ImageSurface(cairo.Format.ARGB32, 0, 0)
+            self._private.handle:render_document(cr, Rsvg.Rectangle {x = 0, y = 0, width = w, height = h})
+        end
     end
 
     if self._private.handle then
-        self._private.handle:render_cairo(cr)
+        cr:set_source_surface(svg_surf, 0, 0)
```

Comment:
This surface has never been drawn to, so what's the point of setting it as a source?

Reply:
Done in https://github.com/awesomeWM/awesome/pull/3881